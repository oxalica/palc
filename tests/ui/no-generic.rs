#[derive(palc::Parser)]
pub struct Parser<T> {
    a: T,
}

#[derive(palc::Args)]
pub struct Args<T> {
    a: T,
}

#[derive(palc::Subcommand)]
pub enum Subcommand<T> {
    A(T),
}

fn main() {}
