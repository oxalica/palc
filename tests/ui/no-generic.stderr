error: generics are not supported
 --> tests/ui/no-generic.rs:1:10
  |
1 | #[derive(palc::Parser)]
  |          ^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `palc::Parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error: generics are not supported
 --> tests/ui/no-generic.rs:6:10
  |
6 | #[derive(palc::Args)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `palc::Args` (in Nightly builds, run with -Z macro-backtrace for more info)

error: generics are not supported
  --> tests/ui/no-generic.rs:11:10
   |
11 | #[derive(palc::Subcommand)]
   |          ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `palc::Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)
